mutiny-core/src/hermes.rs

```diff
@@ -148,26 +151,62 @@ impl<S: MutinyStorage> HermesClient<S> {
         let logger_check_clone = self.logger.clone();
         let stop_check_clone = self.stop.clone();
         let http_client_check_clone = self.http_client.clone();
-        let public_key_check_clone = self.public_key;
+        let nostr_client_check_clone = self.client.clone();
         let base_url_check_clone = self.base_url.clone();
         let current_address_check_clone = self.current_address.clone();
+        let first_federation = self.get_first_federation().await.clone();
         utils::spawn(async move {
             loop {
                 if stop_check_clone.load(Ordering::Relaxed) {
                     break;
                 };
 
-                match check_hermes_name_for_pubkey(
+                match check_hermes_registration_info(
                     &http_client_check_clone,
                     &base_url_check_clone,
-                    public_key_check_clone,
+                    nostr_client_check_clone.clone(),
                 )
                 .await
                 {
                     Ok(o) => {
-                        let mut c = current_address_check_clone.write().await;
-                        log_info!(logger_check_clone, "checked lightning address: {o:?}");
-                        *c = (o, true);
+                        {
+                            let mut c = current_address_check_clone.write().await;
+                            log_info!(
+                                logger_check_clone,
+                                "checked lightning address: {:?}",
+                                o.name
+                            );
+                            *c = (o.name.clone(), true);
+                        }
+
+                        // check that federation is still the same if the user has a name
+                        if o.name.is_some() {
+                            if let Some(f) = first_federation {
+                                // user has a federation, set the current one as the address
+                                match change_federation_info(
```

Comment:
This is going to be called on every start up, should only call `change_federation_info` if the federation ids don't match

Reply:
had that until a did a little refactoring and missed it. added